### PR TITLE
Fix ghost models

### LINKED ISSUE #66: Combining meshes/boolean functions leave ghost models
## Description
Combining multiple meshes leaves ghost models in the Wisp viewport. To understand why this happens it is important to know what goes on behind the scenes. All connections that are made and broken when combining 2 meshes (Cylinder and Torus) are shown [here](https://i.gyazo.com/5060fb93bbd384a4933fab3d6de7cbd0.png)(also shown in *screenshots*).

From this data, we can see that meshes are being disconnected from the original Mesh-node and connected to a PolyUnite-node in an `inputPoly`-array. This seemingly contains all meshes that are combined.
We can conclude from this that meshes are not deleted, but disconnected. Because of this, the mesh parser doesn't get a callback that a mesh is deleted and keeps the meshes in-memory. However, there is a new mesh created which is the combined mesh.

So, a new mesh is created and no meshes are deleted from the scene, leaving a ghost model as all control of the 'old' Mesh-nodes is lost.

### TLDR
No meshes are deleted, just disconnected. This causes Maya to not trigger a 'mesh removed'-callback. The control is lost since 'transform update'-callbacks were bound to the Mesh-node which is not connected anymore. This leaves ghost models.

## To Reproduce
1. Create two (or more) meshes
2. Select both of them in the hierarchy or viewport
3. Go to Mesh -> Combine
4. Move the combined mesh

## Expected behavior
The two transform nodes should be moving one of the combined meshes and there should be only one copy of the combined meshes left.

### Combining meshes
Combining meshes - connection results
**Combining meshes - connection results**

Connections before combining two meshes
**Connections before combining two meshes**

Connections after combining two meshes
**Connections after combining two meshes**

### Boolean function intersection
Intersecting two meshes - connection results
**Intersecting two meshes - connection results**

Connections before intersecting two meshes
**Connections before intersecting two meshes**

Connections after intersecting two meshes
**Connections after intersecting two meshes**

## Additional context
There are a couple of ways to approach this issue. One of them is to just ignore the new mesh and make sure to keep control of the combine meshes individually. Just use the new (combined) mesh its transform as a parent of these two meshes, so the transformations will still be aligned with Maya's default viewport. However, this is pretty much a hack and I'm assuming there are more issues that require fixes like this. So that'll be just a hack, upon hack, upon hack, etc.

The best thing to do is probably reworking the model parser/manager a bit. You basically need it to support grouping as that is what combining does.

### PR DESCRIPTION
**Pull request by MeineZ**
This PR fixes an issue where ghost models appear after combine/boolean operations. It fixes #66.

## Description
Combining and doing boolean operations on meshes caused ghost models to appear. These models weren't controllable anymore by anything, therefore they are ghost models. More information about this in #66.

## Motivation and Context
This change is required to improve the quality of WispForMaya as well as the workflow of artists. Annoying ghost models don't appear any more and undoing these actions don't crash the application anymore either.

## Screenshots:
The fix is shown [here](https://gyazo.com/b981478f8d0e2f0b819f58f8eea992db.gif) in live action.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

**Review by tntmeijs — APPROVED** [review]
Glad you managed to fix this issue. :D

**Review by sainsay — APPROVED** [review]
